src/main/java/com/teragrep/nlf_01/types/ContainerType.java

```diff
@@ -0,0 +1,185 @@
+                .addSDParam(
+                        "enqueued_time", parsedEvent.enqueuedTime() == null ? "" : parsedEvent.enqueuedTime().toString()
+                )
+                .addSDParam("partition_key", partitionKey == null ? "" : partitionKey);
+        parsedEvent.properties().forEach((key, value) -> sdEvent.addSDParam("property_" + key, value.toString()));
+
+        final SDElement sdComponentInfo = new SDElement("aer_02@48577")
+                .addSDParam("timestamp_source", parsedEvent.enqueuedTime() == null ? "generated" : "timeEnqueued");
+
+        final JsonObject mainObject = parsedEvent.asJsonStructure().asJsonObject();
+
+        assertKey(mainObject, "_ResourceId", JsonValue.ValueType.STRING);
+        final ResourceId resourceId = new ResourceId(mainObject.getString("_ResourceId"));
+        final String subscriptionId = resourceId.subscriptionId();
+        final String clusterName = resourceId.resourceName();
+
+        assertKey(mainObject, "PodName", JsonValue.ValueType.STRING);
+        final String podName = mainObject.getString("PodName");
+
+        assertKey(mainObject, "PodNamespace", JsonValue.ValueType.STRING);
+        final String podNamespace = mainObject.getString("PodNamespace");
+
+        assertKey(mainObject, "ContainerId", JsonValue.ValueType.STRING);
+        final String containerId = mainObject.getString("ContainerId");
+
+        final SDElement origin = new SDElement("origin@48577")
+                .addSDParam("subscription", subscriptionId)
+                .addSDParam("clusterName", clusterName)
+                .addSDParam("namespace", podNamespace)
+                .addSDParam("pod", podName)
+                .addSDParam("containerId", containerId);
+
+        assertKey(mainObject, "TimeGenerated", JsonValue.ValueType.STRING);
+        final String timeGenerated = mainObject.getString("TimeGenerated");
+        assertKey(mainObject, "LogSource", JsonValue.ValueType.STRING);
+        final String logSource = mainObject.getString("LogSource");
+        final String logSourceSuffix;
+
+        if ("stdout".equals(logSource)) {
+            logSourceSuffix = ":o";
+        }
+        else if ("stderr".equals(logSource)) {
+            logSourceSuffix = ":e";
+        }
+        else {
+            throw new JsonException("Unknown log source: " + logSource);
+        }
+
+        assertKey(mainObject, "KubernetesMetadata", JsonValue.ValueType.OBJECT);
+        final JsonObject kubernetesMetadata = mainObject.getJsonObject("KubernetesMetadata");
+        assertKey(kubernetesMetadata, "podAnnotations", JsonValue.ValueType.OBJECT);
+        final JsonObject podAnnotations = kubernetesMetadata.getJsonObject("podAnnotations");
+
+        final String hostname = new ASCIIString(
+                podAnnotations.getString(source.source("containerlog.hostname.annotation"))
+        ).withNonAsciiCharsRemoved();
+
+        final String appName = new ASCIIString(
```

Comment:
please ensure the string is a valid appName and does not exceed 48 characters of the rfc5242 specification otherwise throw